tests/MongoDB.Driver.Tests/AuthenticationTests.cs

```diff
@@ -329,20 +374,49 @@ private void AssertAuthenticationFails(MongoClientSettings settings)
         {
             return CreateDatabaseUser(client, "admin", userName, password, "readWriteAnyDatabase", mechanisms);
         }
-        
+
         private BsonDocument CreateDatabaseUser(
             MongoClient client,
             string source,
-            string userName, 
-            string password, 
-            string role, 
+            string userName,
+            string password,
+            string role,
             params string[] mechanisms)
         {
             var createUserCommand =
-                $"{{createUser: '{userName}', pwd: '{password}'," 
+                $"{{createUser: '{userName}', pwd: '{password}',"
                 + $"roles: ['{role}'], mechanisms: {mechanisms.ToJson()}}}";
             return client.GetDatabase(source).RunCommand<BsonDocument>(createUserCommand);
         }
-        
+
+        private BsonDocument CreateX509DatabaseUser(MongoClient client, string userName)
+        {
+            var createUserCommand =
+                $"{{ createUser : '{userName}'," +
+                @"   roles : [ { role : 'readWrite', db : 'test' },
+                               { role : 'userAdminAnyDatabase', db : 'admin' } ]
+                  }";
+            return client.GetDatabase("$external").RunCommand<BsonDocument>(createUserCommand);
+        }
+
+        private void DropDatabaseUser(MongoClient client, string userName, string database)
+        {
+            var dropUserCommand = $"{{ dropUser : '{userName}' }}";
+
+            try
+            {
+                client.GetDatabase(database).RunCommand<BsonDocument>(dropUserCommand);
+            }
+            catch (MongoCommandException) // command throws if user doesn't exist
+            {
+            }
+        }
+
+        private string GetRfc2253FormattedUsernameFromX509ClientCertificate(X509Certificate2 certificate)
+        {
+            var distinguishedName = certificate.SubjectName.Name;
+            var nameWithNoSpaces = string.Join(",", distinguishedName.Split(',').Select(s => s.Trim()));
```

Comment:
Why does removing the spaces matter?

And we're not removing all spaces, just trimming the ones at the beginning and end of each "component" (whatever the right word is for these comma separated values).

I guess I'm just confused at what this method is supposed to do besides replacing "S=" with "ST=".